### PR TITLE
Fix several small inconsistencies with TS deployable units.

### LINKED ISSUE #16420: TS Conyards and MCV don't honor emp-disable
MCV needs 
```yaml
	Mobile:
		PauseOnCondition: empdisable
	Transforms:
		PauseOnCondition: empdisable
```
Conyards need
```yaml
	Transforms:
		PauseOnCondition: being-demolished || empdisable
```

### LINKED ISSUE #16422: Ticktank and artillery can (un)deploy when emp-disabled
They need 
```yaml
	GrantConditionOnDeploy:
		PauseOnCondition: empdisable
```

Same for the Mobile Sensor Array and Juggernaut.

### PR DESCRIPTION
**Pull request by tovl**
split from #16142.

Fixes #16420 
Fixes #16422 

These are some small fixes that where fixed along the way in #16142, but don't depend on any other changes in that PR. Since that PR will probably not be merged for a while, I'd hate for these to be wasted.

### CODE REVIEW

**Comment by matjaeck** [comment]
Is it expected that the order voice plays when deploying tick tanks, artillery, etc? Because it doesn't.

**Comment by tovl** [comment]
Should be fixed now.

File: OpenRA.Mods.Common/Traits/Conditions/GrantConditionOnDeploy.cs [hunk 1]
```diff
@@ -157,6 +159,11 @@ public void ResolveOrder(Actor self, Order order)
 				self.QueueActivity(new DeployForGrantedCondition(self, this));
 		}
 
+		public string VoicePhraseForOrder(Actor self, Order order)
+		{
+			return (order.OrderString == "GrantConditionOnDeploy") ? Info.Voice : null;
```
Comment (pchote) [inline OpenRA.Mods.Common/Traits/Conditions/GrantConditionOnDeploy.cs]: Style nit: The parens aren't needed here.

**Review by pchote — DISMISSED** [review]
LGTM, just one minor code style nit.

Note that the `GrantConditionOnDeploy` change affects several units in other mods, but I consider these extra cases to be bugfixes - e.g. the MAD tank in RA would have an order voice (it was specifically implemented in the `MADTank` trait), but the demo truck did not.

File: OpenRA.Mods.Common/Traits/Conditions/GrantConditionOnDeploy.cs (hunk 1 above)
Comment (tovl) [reply OpenRA.Mods.Common/Traits/Conditions/GrantConditionOnDeploy.cs]: Fixed the style nit and also fixed the same nit in `Transforms`.

Comment (matjaeck) [reply OpenRA.Mods.Common/Traits/Conditions/GrantConditionOnDeploy.cs]: Not fixed yet.

**Review by ghost — COMMENTED** [review]
LGTM

File: OpenRA.Mods.Common/Traits/Conditions/GrantConditionOnDeploy.cs (hunk 1 above)
Comment (matjaeck) [reply OpenRA.Mods.Common/Traits/Conditions/GrantConditionOnDeploy.cs]: Apologize again, another misunderstanding.

**Review by reaperrr — APPROVED** [review]

**Comment by reaperrr** [comment]
https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/b8d645db501bd4fffc767dcd9c22f16b9dd26e07...cf9dc9b87b7fb5b89cdda8f008f7c0af61701349

**Pull request merged**